### DISCUSSION TITLE
void as type

### DISCUSSION THREAD

**Discussion by odussault05** [dotnet/csharplang#8330] [Language Ideas] [answered]
I am sure it is already mention, but why void shouldn't be handled like any type? Is there any drawback? It will be backward compatible since right now we can't do anything with a void. I think it will be a nice syntaxis sugar.

Instead of doing this:
```
public Task Add(MyObj obj)
  {
    _myList.Add(obj); // _myList is List<MyObject>
    return Task.CompletedTask;
  }
```
We would be able to do this one liner:
```
public Task Add(MyObj obj) =>_myList.Add(obj).ToCompletedTask(); // ToCompletedTask is a method extension to return a CompletedTask;
```

And I am sure it will open a whole new world of possibilities.

**Comment by 333fred** [comment, marked as answer]
See https://github.com/dotnet/csharplang/discussions/696.
